src/app/wallet/overview/widgets/coldstake/coldstake.service.ts

```diff
@@ -0,0 +1,128 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+
+import { Log } from 'ng2-logger';
+import { Amount } from '../../../../core/util/utils';
+
+import { RpcService, RpcStateService } from 'app/core/core.module';
+
+@Injectable()
+export class ColdstakeService implements OnDestroy {
+
+  private destroyed: boolean = false;
+  private log: any = Log.create('coldstake-service');
+
+  public coldstake: any =  {
+    txs: [],
+    amount: 0
+  };
+
+  public hotstake: any = {
+    txs: [],
+    amount: 0
+  };
+
+  coldStakingEnabled: boolean = undefined;
+  walletInitialized: boolean = undefined;
+  public encryptionStatus: string = 'Locked';
+
+  private progress: Amount = new Amount(0, 2);
+  get coldstakeProgress(): number { return this.progress.getAmount() }
+  constructor(
+    private _rpc: RpcService,
+    private _rpcState: RpcStateService
+  ) {
+
+    this._rpcState.observe('getwalletinfo', 'encryptionstatus')
+      .takeWhile(() => !this.destroyed)
+      .subscribe(status => {
+        this.encryptionStatus = status;
+        this.rpc_progress();
+      });
+
+    this._rpcState.observe('getwalletinfo', 'txcount')
+      .takeWhile(() => !this.destroyed)
+      .subscribe(txcount => {
+        this.rpc_progress();
+      });
+
+    this._rpcState.observe('getblockchaininfo', 'blocks')
+      .takeWhile(() => !this.destroyed)
+      .subscribe(status => {
+        this.rpc_progress();
+      });
+
+    this._rpcState.observe('ui:coldstaking')
+      .takeWhile(() => !this.destroyed)
+      .subscribe(status => this.coldStakingEnabled = status);
+
+    this._rpcState.observe('ui:walletInitialized')
+      .takeWhile(() => !this.destroyed)
+      .subscribe(status => this.walletInitialized = status);
+    this.rpc_progress();
+  }
+
+  rpc_progress(): void {
+    if (['Unlocked',
```

Comment:
I think this.update() should be called regardless of the state of the wallet (encryptionstatus).

Reply:
The zap shows:

```
*ngIf="_coldstake.hotstake.txs.length
```

Reply:
I agree, will push that